identity/identity_test.go

```diff
@@ -15,3 +20,79 @@ func TestNewIdentity(t *testing.T) {
 	assert.NotEmpty(t, i.Traits)
 	assert.NotNil(t, i.Credentials)
 }
+
+func TestMarshalExcludesCredentials(t *testing.T) {
+	i := NewIdentity(config.DefaultIdentityTraitsSchemaID)
+	credentials := map[CredentialsType]Credentials{
+		CredentialsTypePassword: Credentials{
+			ID: uuid.UUID{},
+		},
+	}
+
+	i.Credentials = credentials
+	jsonBytes, err := json.Marshal(i)
+	assert.Nil(t, err)
+	var jsonMap = map[string]json.RawMessage{}
+	err = json.Unmarshal(jsonBytes, &jsonMap)
+	assert.Nil(t, err)
+	assert.Nil(t, jsonMap["credentials"])
+	assert.Equal(t, credentials, i.Credentials, "Original credentials should not be touched by marshalling")
```

Comment:
You could simply this with:

```suggestion
	var actual bytes.Buffer
	require.Nil(t, json.NewEncoder(&b).Encode(i))
	
	assert.False(t, gjson.Get(b.String(), "credentials").Exists(), "Original credentials should not be touched by marshalling")
	
	// To ensure the original identity is not changed / Unmarshal has no side effects:
	require.NotEmpty(t, i.Credentials)
```